### PR TITLE
(RE-6464) Allow projects to add custom actions to the install

### PR DESCRIPTION
**Pull request by melissa**

### CODE REVIEW

**Comment by jpogran** [comment]
:+1: lgtm

File: resources/windows/wix/project.wxs.erb [hunk 1]
```diff
@@ -55,16 +55,22 @@
     <!-- INSTALLDIR -->
     <CustomAction Id="SaveCmdLineInstallDir" Property="CMDLINE_INSTALLDIR" Value="[INSTALLDIR]" Execute="firstSequence" />
     <CustomAction Id="SetFromCmdLineInstallDir" Property="INSTALLDIR" Value="[CMDLINE_INSTALLDIR]" Execute="firstSequence" />
+    <%= File.read("resources/windows/CustomActions") if File.exist?("resources/windows/CustomActions") -%>
```
Comment (jcoconnor) [inline resources/windows/wix/project.wxs.erb]: @melissa I'm not really comfortable using ERB to read in custom actions.
These should really be specified in a completely separate fragment CA File.

File: resources/windows/wix/project.wxs.erb [hunk 2]
```diff
@@ -55,16 +55,22 @@
...
+    <%- if @platform.architecture == "x86" -%>
+      <%= File.read("resources/windows/x86-CustomActions") if File.exist?("resources/windows/x86-CustomActions") -%>
+    <%- end -%>
 
     <InstallUISequence>
```
Comment (jcoconnor) [inline resources/windows/wix/project.wxs.erb]: We should take this out of the project file altogether and put it in the project specific (e.g. puppet-agent) files.

File: resources/windows/wix/project.wxs.erb (hunk 1 above)
Comment (melissa) [inline resources/windows/wix/project.wxs.erb]: The output looks like https://gist.github.com/melissa/23c6ca022c3aa7375c64.

If you don't like this, by all means give me a better approach

File: resources/windows/wix/project.wxs.erb (hunk 2 above)
Comment (melissa) [inline resources/windows/wix/project.wxs.erb]: cc/ @jpogran

File: resources/windows/wix/project.wxs.erb (hunk 1 above)
Comment (jcoconnor) [inline resources/windows/wix/project.wxs.erb]: NP - will put a concept PR together based on discussions we had with @ferventcoder earlier in week.

**Comment by melissa** [comment]
Closing this, as we've decided on a different approach

File: resources/windows/wix/project.wxs.erb (hunk 1 above)
Comment (ferventcoder) [inline resources/windows/wix/project.wxs.erb]: The erb fragments presented by this approach were what I was hoping for as IMHO the most flexible approach and had discussed with @jamesp.

**Pull request closed without merging**